### PR TITLE
feat(search): update search to DocSearch v3

### PR DESCRIPTION
**Pull request by francoischalifour**
> This PR is a follow-up of [@NataliaTepluhina's tweet](https://twitter.com/N_Tepluhina/status/1290965571023130624).

Hello Vue team! This PR integrates the **new version of DocSearch, that better reflects what we think documentation search should be**.

## Description

Reducing users' efforts to get them started integrating your product is key to adoption. We created [DocSearch](https://docsearch.algolia.com) in 2015 as a community effort to solve this problem.

We've had time to try different approaches since then and would like to propose you a new documentation search experience. This new experience has been well received and is now live on the Docusaurus v2 integration: facebook/docusaurus#2815.

## Preview

Video preview

| Mobile | Desktop |
|-|-|
| Mobile preview | Desktop preview |

## Changes

You can read about all the new features in the Docusaurus PR (facebook/docusaurus#2815), but here's a summary:

- We replaced the dropdown list for a modal
- You can open the search modal with <kbd>Ctrl + K</kbd>, <kbd>⌘ + K</kbd> or <kbd>/</kbd>
- No hard refresh (we leverage the Vue Router navigation API in this integration)
- The experience works on mobile and is close to a native mobile experience
- Recent searches
- Favorite searches
- Selection search
- Search suggestions
- Offline detection

## Notes

I'm not too familiar with VuePress so feel free to use this PR as a base branch, not as-is.

## What's next

This new DocSearch version has been used in production for a couple of weeks now, on the [DocSearch website](http://docsearch.algolia.com/) and on a few [Docusaurus v2](https://v2.docusaurus.io/) integrations.

This version is marked as alpha for now because the underlying libraries that we use (e.g., `autocomplete-core`) are still being worked on at @algolia to make sure that it handles more cases than DocSearch and usual autocomplete search experiences. We plan to go stable in the coming weeks but no major changes will happen to DocSearch.

Let us know what you think!

### CODE REVIEW

File: src/.vuepress/theme/components/AlgoliaSearchBox.vue [hunk 1]
```diff
@@ -36,135 +30,124 @@ export default {
...
+      const url = pathname.replace(this.$site.base, '/') + hash
+
+      return url
+    },
     initialize (userOptions, lang) {
       Promise.all([
-        import(/* webpackChunkName: "docsearch" */ 'docsearch.js/dist/cdn/docsearch.min.js'),
-        import(/* webpackChunkName: "docsearch" */ 'docsearch.js/dist/cdn/docsearch.min.css')
+        import(
+          /* webpackChunkName: "docsearch" */ '../node_modules/@docsearch/js'
```
Comment (francoischalifour) [inline src/.vuepress/theme/components/AlgoliaSearchBox.vue:45]: The build system doesn't seem to accept external imports starting with `@` so I referenced to `node_modules` here, but it doesn't build. Feel free to update this 🙂

File: src/.vuepress/theme/components/AlgoliaSearchBox.vue [hunk 2]
```diff
@@ -36,135 +30,124 @@ export default {
...
+            {
+              placeholder: this.$site.themeConfig.searchPlaceholder
+            },
+            userOptions,
+            {
+              container: '#docsearch',
+              // #697 Make docsearch work well at i18n mode.
+              searchParameters: Object.assign(
+                {},
+                // lang && {
```
Comment (francoischalifour) [inline src/.vuepress/theme/components/AlgoliaSearchBox.vue:64]: Not sure yet how you plan to use these filters on the new documentation.

**Comment by NataliaTepluhina** [comment]
@francoischalifour thank you, this looks amazing! Let me take a look at how to fix the build issue

**Comment by NataliaTepluhina** [comment]
@francoischalifour sorry, it did take some time to figure it out! In order to make it work on Netlify build, we need to:

- move all dependencies to the root `package.json`
- change paths for `@docsearch` from relative to absolute ones:

```js
        import(/* webpackChunkName: "docsearch" */ '@docsearch/js'),
        import(/* webpackChunkName: "docsearch" */ '@docsearch/css')
```

I've created a PR to test the deployment: https://github.com/vuejs/docs-next/pull/410/files but I would be happy to close this if you could add requested changes to this PR 😅 

Thank you one more time for creating it! Search works and looks amazing

**Comment by francoischalifour** [comment]
Thank you for investigating @NataliaTepluhina. Build issue is addressed in a6d5adf810a8bce75e23c8d17eff5e88a6bd79cb.

File: src/.vuepress/theme/components/AlgoliaSearchBox.vue (hunk 1 above)
Comment (francoischalifour) [reply src/.vuepress/theme/components/AlgoliaSearchBox.vue]: Fixed in a6d5adf810a8bce75e23c8d17eff5e88a6bd79cb.

**Comment by NataliaTepluhina** [comment]
Thank you @francoischalifour! Happy to merge 🎉

**Comment by meteorlxy** [comment]
Have implemented this PR in vuepress 2, as `@vuepress/plugin-docsearch`

> https://github.com/vuepress/vuepress-next/commit/ae5e45a157ecfed83401df64ca30ecffc5045341

**Pull request merged**